### ISSUE TITLE
Entity beginning with digit creates error in parsing

### ISSUE THREAD

**Issue by AdrienHorgnies** [jhipster/generator-jhipster#5911]
##### **Overview of the issue**

Creating an entity with its name beginning with a number will result in gulp shouting errors.
```
yo jhipster:entity 03Sweet
[...lots of skipped lines...]
Running `gulp inject` to add JavaScript to index.html

[12:30:07] Using gulpfile ~/JHipster/samples_apps/entityWithNumber/gulpfile.js
[12:30:07] Starting 'inject:app'...
[12:30:07] gulp-notify: [JHipster Gulp Build] Error: Error in parsing: "entities/03-sweet/03-sweet.controller.js", Line 6: Unexpected token ILLEGAL
[12:30:07] gulp-notify: [JHipster Gulp Build] Error: Error in parsing: "entities/03-sweet/03-sweet.service.js", Line 5: Unexpected token ILLEGAL
[12:30:07] gulp-notify: [JHipster Gulp Build] Error: Error in parsing: "entities/03-sweet/03-sweet.state.js", Line 53: Unexpected token ILLEGAL
[12:30:07] gulp-notify: [JHipster Gulp Build] Error: Error in parsing: "entities/03-sweet/03-sweet-delete-dialog.controller.js", Line 6: Unexpected token ILLEGAL
[12:30:07] gulp-notify: [JHipster Gulp Build] Error: Error in parsing: "entities/03-sweet/03-sweet-detail.controller.js", Line 6: Unexpected token ILLEGAL
[12:30:07] gulp-notify: [JHipster Gulp Build] Error: Error in parsing: "entities/03-sweet/03-sweet-dialog.controller.js", Line 6: Unexpected token ILLEGAL
[12:30:08] gulp-inject Nothing to inject into index.html.
[12:30:08] Finished 'inject:app' after 258 ms
```

##### **Motivation for or Use Case**

It should be clear whether such input is accepted or not. Either validation should refuse the input or entity generator should support such input.

It's a valid value for [mysql](https://dev.mysql.com/doc/refman/5.7/en/identifiers.html) : 
* Identifiers may begin with a digit but unless quoted may not consist solely of digits.

It's may more be a question of convention tough.

##### **Reproduce the error**

`yo jhipster:entity 03Sweet`
Or anything beginning with digits.

##### **Related issues**

None

##### **Suggest a Fix**

If it must be refused by validation, I can add the additional validation rule at [the pertinent place](https://github.com/jhipster/generator-jhipster/blob/master/generators/entity/index.js#L180-L190).

##### **JHipster Version(s)**

Bug occured as of **generator-jhipster@4.3.0**

- [x] Checking this box is mandatory (this is just to show you read everything)

**Comment by pascalgrimaud** [comment]
I confirm the issue, and I'm surprised that we have this since a long time!
Would you like to PR a fix?

I tested the JDL Studio and I got: `Expected ['\n'|'\t'|'\r'|' '| | ] or [A-Z] but "0" found. -> line: 1` so the control is already done there

**Comment by AdrienHorgnies** [comment]
```javascript
else if ((/^[0-9].*$/.test(this.name))) {
                this.error(chalk.red('The entity name cannot start with a number'));
}
```

I'll test and PR this fix tomorrow morning.

**Comment by AdrienHorgnies** [comment]
Can someone check it ? https://github.com/jhipster/generator-jhipster/pull/5917

**Comment by deepu105** [comment]
closed via #5917
